internal/graph/dotgraph.go

```diff
@@ -127,7 +128,7 @@ func (b *builder) addLegend() {
 	}
 	title := labels[0]
 	fmt.Fprintf(b, `subgraph cluster_L { "%s" [shape=box fontsize=16`, title)
-	fmt.Fprintf(b, ` label="%s\l"`, strings.Join(labels, `\l`))
+	fmt.Fprintf(b, ` label="%s\l"`, html.EscapeString(strings.Join(labels, `\l`)))
```

Comment:
It's not obvious to me why `html.EscapeString()` is used.

This code generates a dot file, this has nothing to do with HTML. I'd expect this to adhere to whatever string quoting / special character rules for strings in dot are. And https://graphviz.org/doc/info/lang.html says:

> In quoted strings in DOT, the only escaped character is double-quote ("). That is, in quoted strings, the dyad \" is converted to "; all other characters are left unchanged. In particular, \\ remains \\. Layout engines may apply additional escape sequences.

Plus see escaping rules for escString in https://graphviz.org/doc/info/attrs.html.

So I think we should replace "\" with "\\" and `"` with `\"`, am I missing something?

Please also add a test.

Reply:
You're right, I was initially under the impression that graphviz runs into issues with non-html characters but this is purely an issue with the dot file. Amended the commit, added a test for the escapeDoubleQuotes function, and added labels with quotes to the test files.

Reply:
Just noticed a formatting issue in the string, will amend it momentarily.

Reply:
...done.